pkg/apiserver/handlers.go

```diff
@@ -218,57 +237,89 @@ func WithAuthorizationCheck(handler http.Handler, getAttribs RequestAttributeGet
 // Fully qualified paths for above:
 // /api/{version}/*
 // /api/{version}/*
-func KindAndNamespace(req *http.Request) (namespace, kind string, parts []string, err error) {
-	parts = splitPath(req.URL.Path)
-	if len(parts) < 1 {
-		err = fmt.Errorf("Unable to determine kind and namespace from an empty URL path")
-		return
+func GetAPIRequestInfo(req *http.Request) (APIRequestInfo, error) {
```

Comment:
This needs to be a method on something that is holding on to apiPrefixes (as per no statics)